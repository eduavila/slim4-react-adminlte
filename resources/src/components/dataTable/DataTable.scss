.datatable{
    width: 100%;
    
    .no-data-content{
        margin: 0 auto;
        padding-top: 40px;
        max-width: 400px;
        text-align: center;

        h2{
            font-size: 1.1rem;
        }
    }

    .table-responsive{
        border:none !important;
    }

    .react-bootstrap-table-pagination{
        margin-right: 0px;
        margin-left: 0px;
    }

    .progress-bar span{
        color:#373A3C;
    }

    .reset-expansion-style{
        padding:0px;
    }

    // .expand-cell-header{
    //     width: 40px !important;
    // }
    .expand-cell{
        width: 40px !important;
    }

    .pagination>li>a{
        padding: 1px 9px;
    }
    .row-action > div{
        margin: 0px;
    }
    
}

.search-label{
    width: 100%;
}


.col-fixed-left {
    position: sticky !important;
    left: 0;
    z-index: 0;
    background: white;
}

// TABLE VERYCONDESEND

.table-sm>tbody>tr>td,.table-sm>tbody>tr>th,.table-sm>tfoot>tr>td,.table-sm>tfoot>tr>th,.table-sm>thead>tr>td,.table-sm>thead>tr>th {
    padding: 3px;
    font-size: 12px;
}

.table tbody{
    tr > td{
        position: relative;
    }
}

.row-action > div{
    margin-bottom: 0px;
}