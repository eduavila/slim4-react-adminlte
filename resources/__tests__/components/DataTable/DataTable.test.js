import React from 'react';
import { shallow,mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import DataTable from '../../../src/components/dataTable';

const colums = [
    {
        dataField: 'nome',
        text: 'Arquivo',
        headerStyle:{
            textAlign:'center'
        },
        style: {
            width: '50px',
            textAlign:'center'
        },
        sort: true
    },
    {
        dataField: 'tarefa_id',
        text: 'Tarefa',
        sort: true,
        headerStyle:{
            textAlign:'center'
        },
        style: {
            width: '100px'
        }
    }
]

describe('<DatTable />', () => {
    describe('render()', () => {
        test('renders the component', () => {
            const wrapper = shallow(<DataTable columns={ colums } />);
        
            expect(toJson(wrapper)).toMatchSnapshot();
        });

        test('Deve verificar se criou tr corretamente', () => {
            const data = [
                {
                    id: 1,
                    tarefa_id: 1,
                    nome: 'Arquivos.jpg'
                },
                {
                    id: 2,
                    tarefa_id: 2,
                    nome: 'Arquivos2.jpg'
                }
            ]
            const wrapper = mount(<DataTable columns={ colums } data={data} />);

            expect(wrapper.find('tbody').find('tr').length).toEqual(2);
        });

        test('Deve verificar label de sem dados', () => {
            const data = []
            const wrapper = mount(<DataTable columns={ colums } data={data} />);

            expect(wrapper.find('.text-center').text().trim()).toEqual('Nenhum registro.');
        });

        test('Deve verificar loader carregamento', () => {
            const data = []
            const wrapper = mount(<DataTable columns={ colums } data={data}  loading={true}/>);

            expect(wrapper.find('.ui.loader').length).toEqual(1);
        });

        test('Deve verificar botoes de ações', () => {
            const data = [];
            const actions = () => (<a href="#" id="btn-acoes" className="btn btn-sm btn-primary btn-flat pull-left">Nova Ação</a>);
            const wrapper = mount(<DataTable columns={ colums } data={data} actions={ actions }/>);

            expect(wrapper.find('#btn-acoes').first().text()).toEqual('Nova Ação');
        });
    });
});