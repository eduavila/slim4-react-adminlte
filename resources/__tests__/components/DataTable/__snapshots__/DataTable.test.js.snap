// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DatTable /> render() renders the component 1`] = `
<div>
  <Toolkitprovider
    columns={
      Array [
        Object {
          "dataField": "nome",
          "headerStyle": Object {
            "textAlign": "center",
          },
          "sort": true,
          "style": Object {
            "textAlign": "center",
            "width": "50px",
          },
          "text": "Arquivo",
        },
        Object {
          "dataField": "tarefa_id",
          "headerStyle": Object {
            "textAlign": "center",
          },
          "sort": true,
          "style": Object {
            "width": "100px",
          },
          "text": "Tarefa",
        },
      ]
    }
    data={Array []}
    keyField="id"
    search={
      Object {
        "onColumnMatch": [Function],
      }
    }
  >
    <Component />
  </Toolkitprovider>
  <p
    className="text-center"
  >
     Nenhum registro. 
  </p>
</div>
`;
